modules/migrations/update.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"strconv"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+)
+
+// UpdateRepoMigrations will update posterid on issues/comments/prs when external login user
+// login or when migrating
+func UpdateRepoMigrations(repoID, externalUserID, userID int64) error {
+	if err := models.UpdateIssuesMigrations(repoID, externalUserID, userID); err != nil {
+		return err
+	}
+
+	return models.UpdateCommentsMigrations(repoID, externalUserID, userID)
+}
+
+// UpdateMigrationPosterID updates all migrated repositories' issues and comments posterID
+func UpdateMigrationPosterID() {
+	if err := updateMigrationPosterIDByGitService(models.GithubService); err != nil {
+		log.Error("updateMigrationPosterIDByGitService failed: %v", err)
+	}
+}
+
+func updateMigrationPosterIDByGitService(tp models.GitServiceType) error {
+	provider := tp.Name()
+	if len(provider) == 0 {
+		return nil
+	}
+
+	var repoStart int
+	const batchSize = 100
+	for {
+		ids, err := models.FindMigratedRepositoryIDs(tp, batchSize, repoStart)
+		if err != nil {
+			return err
+		}
+
+		var start int
+		for {
+			users, err := models.FindExternalUsersByProvider(models.FindExternalUserOptions{
+				Provider: provider,
+				Start:    start,
+				Limit:    batchSize,
+			})
+			if err != nil {
+				return err
+			}
+
+			for _, user := range users {
+				externalUserID, err := strconv.ParseInt(user.ExternalID, 10, 64)
+				if err != nil {
+					log.Warn("Parse externalUser %#v 's userID failed: %v", user, err)
```

Comment:
Perhaps this user should never be attempted again?

Reply:
@lunny May I ask why `strconv.ParseInt` is required here? both `externalUserID` and `user.ExternalID` has a `string` type, and some OpenID provider like `keycloak` uses UUID string for JWT `sub` field.

And I got an error:
```
2019/10/16 20:46:48 routers/user/auth.go:807:LinkAccountPostSignIn() [E] UserLinkAccount: strconv.ParseInt: parsing "a9ba181f-3b75-4c3a-a571-47f9d2cb6b5c": invalid syntax
```

Reply:
@vizv Please could you open a new issue. Your problem is very likely to go missing in such a busy already merged PR.

Reply:
@zeripath Thanks, I'll trying to fix it myself and probably open a PR after work.